tools/lsp-wake/astree.cpp

```diff
@@ -437,10 +482,28 @@ std::string ASTree::sanitizeComment(std::string comment) {
   return comment;
 }
 
+std::string ASTree::composeOuterComment(std::vector<std::pair<std::string, int>> comment) {
+  std::string composed;
+  for (int i = (int) comment.size() - 2; i >= 0; i--) {
```

Comment:
So, the reason you get a warning here is that `size()` might not fit in an `int`. You're probably correct, and it will fit, so the override is ok. I generally do this sometimes, too. However, it is usually better to use a `size_t` to be safe (also the compiler likes these better when optimizing). Unfortunately, `size_t` cannot accommodate negative numbers.

I would probably leave the code how you wrote it, but wanted to point out the pitfalls of the cast.

Reply:
Yes, I understand. Left as is